Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -371,6 +380,9 @@ void RenderBox::styleDidChange(StyleDifference diff, const RenderStyle* oldStyle
     if (isDocElementRenderer || isBodyRenderer) {
 #if ENABLE(DARK_MODE_CSS)
         view().frameView().recalculateBaseBackgroundColor();
+
+        if (oldStyle && oldStyle->colorScheme() != newStyle.colorScheme())
+            view().frameView().invalidateScrollCorner(view().frameView().scrollCornerRect());
```

Comment:
Do we also need to `invalidateScrollbars()` here? If so, it might be nice to factor:

```
scrollableArea->invalidateScrollCorner(scrollableArea->scrollCornerRect());
scrollableArea->invalidateScrollbars();
```

into a method.

Reply:
Also, is this code necessary given the added logic above? The document element renderer should have a layer, but maybe the body does not? Want to make sure this isn't redundant.

Reply:
Viewport scrollbars already correctly invalidate (not sure where that happens perhaps as a side effect rather than a direct function call), so we don't need invalidateScrollbars for this block. 

We do however need the scroll corner invalidate because that's not handled elsewhere nor by the above code (I'm rebuilding locally just to double check this but I'm 99% sure)

Reply:
I wonder if we need to make `LocalFrameView::invalidateScrollbarsForAllScrollableAreas()` also invalidate the scroll corner...

Reply:
I'll take a look at what that will do. It might remove the neccessity to do the block for documents here.

Reply:
It doesn't allow me to remove any of the code in RendeBox (Which is all neccessary) but it does fix the issue with  scroll corner changing when "color-scheme: light dark" is specified and the scroll corner doesn't repaint. So I've added that change will push it up now.